python/ql/test/library-tests/regexparser/Consistency.ql

```diff
@@ -0,0 +1,15 @@
+/**
+ * Flags regular expressions that are parsed ambigously
+ */
+
+import python
+import semmle.python.RegexTreeView
+
+from string str, int counter, Location loc
+where
+  counter =
+    strictcount(RegExpTerm term |
+      term.getLocation() = loc and term.isRootTerm() and term.toString() = str
```

Comment:
The two tests operate on different levels: `Regex`, which is a string decorated with some parsing methods, versus `RegexTerm` which is an IPA type encoding a parse tree. Only the latter has the concept of `isRootTerm`.

It might still be a good idea to create a query predicate and import it into various places where we have regexes in code (rather than, as we do now, import all those .py-files into the consistency check folder).